data_steward/cdr_cleaner/cleaning_rules/unit_normalization.py

```diff
@@ -0,0 +1,302 @@
+  unit_source_value,
+  value_source_value
+FROM
+  `{{project_id}}.{{dataset_id}}.{{measurement_table}}`
+LEFT JOIN
+  `{{project_id}}.{{dataset_id}}.{{unit_table_name}}`
+USING
+  (measurement_concept_id,
+    unit_concept_id)""")
+
+
+class UnitNormalization(BaseCleaningRule):
+    """
+    Units for labs/measurements will be normalized..
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'Units for labs/measurements will be normalized using unit_mapping.csv.'
+        super().__init__(issue_numbers=['DC414', 'DC700'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.DEID_CLEAN],
+                         affected_tables=['measurement'],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_mapping_list(self, unit_mappings_list):
+        """
+        Filters out name columns from unit_mappings.csv file and returns list of mappings suitable for BQ
+
+        :param unit_mappings_list:
+        :return: formatted list suitable for insert in BQ:
+            (measurement_concept_id, unit_concept_id, set_unit_concept_id, transform_value_as_number)
+        """
+        pair_exprs = []
+        for route_mapping_dict in unit_mappings_list:
+            pair_expr = '({measurement_concept_id}, {unit_concept_id}, {set_unit_concept_id}, ' \
+                        '\"{transform_value_as_number}\")'.format(**route_mapping_dict)
+            pair_exprs.append(pair_expr)
+        formatted_mapping_list = ', '.join(pair_exprs)
+        return formatted_mapping_list
+
+    def setup_rule(self, client):
+        """
+        Load required resources prior to executing cleaning rule queries.
+
+        Method to run data upload options before executing the first cleaning
+        rule of a class.  For example, if your class requires loading a static
+        table, that load operation should be defined here.  It SHOULD NOT BE
+        defined as part of get_query_specs().
+        :param client:
+        :return:
+        """
+
+        # creating _unit_mapping table
+        unit_mapping_table = f'{self.get_project_id()}.{self.get_dataset_id()}.{UNIT_MAPPING_TABLE}'
+        try:
+            bq.create_tables(client, self.get_project_id(),
+                             [unit_mapping_table])
+            # Uploading data to _unit_mapping table
+            unit_mappings_csv_path = os.path.join(resources.resource_files_path,
+                                                  UNIT_MAPPING_FILE)
+            result = bq.upload_csv_data_to_bq_table(
```

Comment:
This result should check for an error property on the result.  The errors are not automatically raised.  This is the [LoadJob](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.job.LoadJob.html) documentation.  This is a [stack overflow answer](https://stackoverflow.com/questions/57039771/how-to-get-detailed-big-query-error-by-using-python) showing how it is used.  We should either do this here or possibly in the utility you have created.